apps/src/codebridge/InfoPanel/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState} from 'react';
+
+import Button from '@cdo/apps/componentLibrary/button';
+import InstructionsView from '@cdo/apps/lab2/views/components/Instructions';
+import PanelContainer from '@cdo/apps/lab2/views/components/PanelContainer';
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import HelpAndTips from './HelpAndTips';
+
+import moduleStyles from './styles/info-panel.module.scss';
+
+enum Panels {
+  Instructions = 'Instructions',
+  HelpAndTips = 'Help and Tips',
+}
+
+export const InfoPanel = React.memo(() => {
+  const mapReference = useAppSelector(
+    state => state.lab.levelProperties?.mapReference
+  );
+  const referenceLinks = useAppSelector(
+    state => state.lab.levelProperties?.referenceLinks
+  );
+  const [currentPanel, setCurrentPanel] = useState(Panels.Instructions);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const renderHeaderButton = () => {
+    return panelOptions().length > 1 ? (
+      <div>
+        <Button
+          icon={{
+            iconStyle: 'solid',
+            iconName: isDropdownOpen ? 'caret-up' : 'caret-down',
+          }}
+          isIconOnly
+          onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+          color={'black'}
+          ariaLabel={'Information panel dropdown'}
+          size={'xs'}
+        />
+      </div>
+    ) : null;
+  };
+
+  const panelOptions = () => {
+    // For now, always include Instructions panel.
+    // TODO: support hiding this panel completely if there are no instructions.
+    const panelOptions = [Panels.Instructions];
+    if (mapReference || referenceLinks) {
+      panelOptions.push(Panels.HelpAndTips);
+    }
+    return panelOptions;
+  };
+
+  const showCurrentPanel = () => {
+    switch (currentPanel) {
+      case Panels.Instructions:
+        return <InstructionsView />;
+      case Panels.HelpAndTips:
+        return <HelpAndTips />;
+    }
+  };
+
+  const changePanel = (panel: Panels) => {
+    setCurrentPanel(panel);
+    setIsDropdownOpen(false);
+  };
+
+  return (
+    <PanelContainer
+      id="codebridge-info-panel"
+      headerContent={currentPanel}
+      rightHeaderContent={renderHeaderButton()}
+      className={moduleStyles.infoPanel}
+    >
+      {isDropdownOpen && (
+        <form className={moduleStyles.dropdownContainer}>
+          <ul>
+            {panelOptions().map(panel => {
+              return (
```

Comment:
Nit: If this map function just returns the list item, I think we can actually remove the function's outer curly brackets and explicit `return` keyword.

Reply:
I get errors when I do that. I believe the reason is this is inline code rather than a function, and any code inside the html needs to be surrounded by curly braces.

Reply:
What errors are you getting? I was curious and tried modifying locally and it seems to work. You should be able to remove the curly braces and `return` - and the return statement's `;`, which I missed initially!
I believe you only need the curly braces and return if the arrow function includes other steps.